botocore/endpoint.py

```diff
@@ -314,6 +314,11 @@ def _do_get_response(self, request, operation_model, context):
             response_dict, operation_model.output_shape
         )
         parsed_response.update(customized_response_dict)
+        if updated_status_code := customized_response_dict.get(
+            'updated_status_code'
+        ):
+            http_response.status_code = updated_status_code
+            del parsed_response['updated_status_code']
```

Comment:
This feels kind of awkard; if we have to do this, we can move this added section one line up and delete it from the customized response dict instead.  

Better yet would be updating the ResponseMetadata.HTTPStatusCode in the customized response dictionary, then checking the parsed response in the future handler.  Not sure if this is possible without trying it out.